src/components/autocomplete/js/autocompleteController.js

```diff
@@ -684,6 +686,20 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
   }
 
   /**
+   * Removes the given term after the specified cache lifetime
+   * @param term
+   */
+  function removeAfterLifetime(term) {
+    if ($scope.cacheLifetime && lifetimeQueue.indexOf(term) === -1) {
+      lifetimeQueue.push(term);
+      $timeout(function() {
+        cache[term] = null;
+        lifetimeQueue.splice(lifetimeQueue.indexOf(term), 1);
```

Comment:
The thing is, the index will change if a previous queue task will remove an item from the list.

So we need to retrieve the index on removing.
I think indexOf is a good solution.